### ISSUE TITLE
New ecosystem: Leiningen

### ISSUE THREAD

**Issue by colinphill-mdsol** [dependabot/dependabot-core#10715]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

I would like to offer to develop and contribute support for Leiningen, a major build tool for Clojure, to dependabot-core.

Leiningen is closely related to Maven – it uses a similar declarative style and integrates with Maven artifact repositories – differing primarily in syntax. Most of the work is already done, therefore, by the Maven support already included in Dependabot, which can serve as a model for the new ecosystem. Leiningen has built-in tools for programmatically changing its dependency file, so the significant syntactic difference will not result in significant additional complexity.

There are [47.2k public projects](https://github.com/search?q=path%3A%2F.*%5C%2Fproject%5C.clj%24%2F++NOT+is%3Aarchived&type=code) using Leiningen, plus some unknown number of private projects (my company has a few dozen). This is a bit more than [Hex's 30.3k public projects](https://github.com/search?q=path%3A%2F.*%5C%2Fmix.exs%24%2F++NOT+is%3Aarchived&type=code) and not terribly far behind [Gradle's 61.4k public projects](https://github.com/search?q=path%3A%2F.*%5C%2Flibs%5C.versions%5C.toml%24%2F+++NOT+is%3Aarchived&type=code), both of which are currently included as built-in ecosystems.

Leiningen is currently the poster child for extending Dependabot with custom scripts, being [documented in CONTRIBUTING.md](https://github.com/dependabot/dependabot-core/blob/654de3de33edce7a9e202a5e77a52891d60f7f3c/CONTRIBUTING.md#whats-next) as an example to follow. However, [the runner linked there](https://github.com/CGA1123/dependabot-lein-runner) is built on top of a [fork](https://github.com/CGA1123/dependabot-core/tree/leiningen) which is over 9000 commits behind main. It was last updated back in 2020, [before you revamped Dependabot's extensibility](https://github.com/dependabot/dependabot-core/pull/2769#issuecomment-748130990).

**Comment by colinphill-mdsol** [comment]
That automatic label doesn't seem quite correct.

**Comment by colinphill-mdsol** [comment]
Just confirming that six months later, I am still available to develop and support this, and it would still be very valuable to my team and to the Clojure community.

**Comment by D3bu663r** [comment]
Hey @colinphill-mdsol , I just opened a [PR](https://github.com/dependabot/dependabot-core/pull/12710) with a new attempt to add support for Clojure.
It's still in an early stage, and I'm currently waiting on feedback from the team to see if this is something worth moving forward with.

If you believe it makes sense, I'd be happy to collaborate with you on this implementation!

**Comment by colinphill-mdsol** [comment]
@D3bu663r Nice! I'll subscribe to your PR. Hopefully the Dependabot team finds the time to follow up on it.

**Comment by colinphill-mdsol** [comment]
Confirming again that I am still available to develop and maintain this — perhaps in collaboration with @D3bu663r — and that the labels on this issue are incorrect. I have not begun work because I haven't gotten any word on whether the Dependabot team would accept the contribution.
